spec/e2e/relationship/persistence/query_factory_spec.rb

```diff
@@ -103,6 +103,59 @@ def self.count
         expect(from_node.reload.to_classes).to be_empty
       end
 
+      it 'does not create duplicate has_one relationship' do
+        from_node.save
+        to_node.save
+        begin
+          $concurrency_test = true
+          $concurrency_queue = Queue.new
+          $concurrency_test_wait = true
+          t1 = Thread.new { to_node.update(from_class: from_node) }
+          t2 = Thread.new { to_node.update(from_class: from_node) }
+          while $concurrency_queue.size < 2
+            # wait till both thread have complted read query
+            sleep 1
+          end
+          $concurrency_test_wait = false # make threads resume their work
+          [t1, t2].join # wait for the threads to finish
+
+          # to_node.from_class only returns 1 result but in db there are two rels associated with this node
+          # this assertion fails with with result 2 proving the presence of bug
+          expect(ActiveGraph::Base.query("MATCH (node2:`ToClass`)<-[rel1:`HAS_REL`]-(from_class:`FromClass`) return from_class").to_a.size).to eq(1)
```

Comment:
Screenshot of the data
<img width="254" alt="Screenshot 2023-11-30 at 8 09 28 PM" src="https://github.com/neo4jrb/activegraph/assets/9930523/3c82df4d-1b69-4952-8762-0fea9fa4716d">